### PR TITLE
rc: Use $PWD instead of $HOME to allow relative sourcing from anywhere

### PR DESCRIPTION
**Pull request by cjsut**
This PR updates the `rc` file to determine non-relative path using `$PWD` instead of `$HOME`, which previously required the user to be in `$HOME` and assumed the file was linked or present in that directory.

### CODE REVIEW

File: rc [hunk 1]
```diff
@@ -9,19 +9,20 @@
...
-            BASHCONF_CONFIG_DIR="${BASHCONF_CONFIG_DIR%/*}"
-	done
+        # Get the (actual) location of this file.
+        test -L "$BASH_SOURCE" \
+            && BASHCONF_CONFIG_DIR="$(readlink "$BASH_SOURCE")" \
+            || BASHCONF_CONFIG_DIR="$BASH_SOURCE"
+
+        # Ensure we have an absolute path.
+        test "${BASHCONF_CONFIG_DIR:0:1}" = '/' \
+            || BASHCONF_CONFIG_DIR="$PWD/$BASHCONF_CONFIG_DIR"
```
Comment (cjsut) [inline rc:19]: There are probably still some edge cases here. We're assuming that IF it's not an absolute path, it's sourced relative to the current directory; that sounds right, but I'm a little rusty.

Comment (cjsut) [reply rc:19]: Let's go with it. Unlikely anyone else is using this anyway (and if you are, let me know!)
